core/src/main/java/org/apache/iceberg/ManifestFilterManager.java

```diff
@@ -77,9 +78,12 @@ public String partition() {
   private boolean failMissingDeletePaths = false;
   private int duplicateDeleteCount = 0;
   private boolean caseSensitive = true;
+  private boolean allDeletesReferenceManifests = true;
+  private boolean useReferencedManifests = false;
 
   // cache filtered manifests to avoid extra work when commits fail.
   private final Map<ManifestFile, ManifestFile> filteredManifests = Maps.newConcurrentMap();
+  private final Set<String> manifestsReferencedForDeletes = Sets.newHashSet();
```

Comment:
We already have these two vars are that are closely related.

```
private final PartitionSet deleteFilePartitions;
private final Set<F> deleteFiles = newFileSet();
```

I would personally call this `deleteFileManifests` and co-locate them for consistency.

Reply:
I co-located the fields, I'm still not sure I agree with the name `deleteFileManifests` because it's more than just delete file manifests that will be stored in the set; it can technically also be data file manifests as well. That's why I think `manifestsWithDeletedFiles` is a more accurate name.